File: MySpace/wwwroot/TestScreenOCR/MySpace/js/jsfunctions/registerUser.js
function registerUser() {

    // -------- Collect Form Data --------
    const data = {
        FirstName: $("input[placeholder='John']").val().trim(),
        LastName: $("input[placeholder='Doe']").val().trim(),
        Email: $("input[type='email']").val().trim(),
        Username: $("input[placeholder='username']").val().trim(),
        Password: $("#pwd").val(),
        ConfirmPassword: $("#cpwd").val()
    };

    // -------- Basic Required Field Validation --------
    if (!data.FirstName || !data.LastName || !data.Email ||
        !data.Username || !data.Password) {
        alert("All fields are required");
        return;
    }

    // -------- Password Length Validation --------
    if (data.Password.length < 8) {
        alert("Password must be at least 8 characters");
        return;
    }

    // -------- Password Match Validation --------
    if (data.Password !== data.ConfirmPassword) {
        alert("Passwords do not match");
        return;
    }

    // -------- AJAX Call : Register User --------
    $.ajax({
        url: "/Home/RegisterUser",
        type: "POST",
        contentType: "application/json",
        data: JSON.stringify(data),
        success: function (res) {
            if (res.success) {
                alert(res.message);
                window.location.href = "/Home/MySpace_Login";
            } else {
                alert(res.message);
            }
        },
        error: function () {
            alert("Server error. Please try again.");
        }
    });
}

File: MySpace/wwwroot/TestScreenOCR/MySpace/Controller/Controllerfunctions/RegisterUser.cs
        public async Task<IActionResult> RegisterUser([FromBody] RegisterVM model)
        {
            // -------- Model Validation --------
            if (!ModelState.IsValid)
                return Json(new { success = false, message = "Invalid input data" });

            // -------- Password Match Validation --------
            if (model.Password != model.ConfirmPassword)
                return Json(new { success = false, message = "Passwords do not match" });

            // -------- Call DAL to Register User --------
            bool result = await _dal.RegisterUserAsync(
                model.FirstName,
                model.LastName,
                model.Email,
                model.Username,
                model.Password
            );

            // -------- User Already Exists --------
            if (!result)
                return Json(new { success = false, message = "User already exists" });

            // -------- Registration Success --------
            return Json(new
            {
                success = true,
                message = "Account created successfully"
            });
        }